unittests/savanna_proposer_policy_tests.cpp

```diff
@@ -234,6 +234,7 @@ BOOST_FIXTURE_TEST_CASE(no_proposer_policy_change_without_finality_2, savanna_cl
 // ---------------------------------------------------------------------------------------------------
 BOOST_FIXTURE_TEST_CASE(pending_proposer_policy_becomes_active_without_finality, savanna_cluster::cluster_t) try {
    auto& A=_nodes[0];
+   static_assert(prod_rep >= 4);
```

Comment:
This should be `static_assert(prod_rep > 4);` since below you have:
```cpp
while (A.head().timestamp().slot % prod_rep >= prod_rep - 4)
   A.produce_block();
```

Note that if `prod_rep == 4`, then it would get stuck in an infinite loop.

Better yet, you can keep the `static_assert` as is and instead fix the condition in the `while` loop so that it reflects the comment properly. (See first sentence in https://github.com/AntelopeIO/spring/pull/607#discussion_r1725796947.)

For example:
```cpp
while ((A.head().timestamp().slot + 1) % prod_rep >= prod_rep - 3)
   A.produce_block();
```

Reply:
Thanks again, done!